src/util-memcmp.c

```diff
@@ -71,10 +65,10 @@ static int MemcmpTest04 (void)
     uint8_t a[] = "abcd";
     uint8_t b[] = "abcD";
 
-    int r = SCMemcmp(a, b, sizeof(a)-1);
+    int r = SCMemcmp(a, b, sizeof(a) - 1);
     if (r != 1) {
         printf("%s != %s, but memcmp returned %d: ", a, b, r);
-        return 0;
+        FAIL;
     }
```

Comment:
Here, the new approach is:
`FAIL_IF(r != 1);`

(You could also, alternatively, use `FAIL_IF_NOT(r == 1);` )

We don't need the print statement - the test will fail immediately, so we will know which check failed. And we want to get rid of all `if` statements.